### ISSUE TITLE
v1 documentation

### ISSUE THREAD

**Issue by LeviSchuck** [kbrsh/moon#134]
Until all the beautiful docs are updated, I figure it would be nice to have a place where the core breaking changes are documented.

So far from looking at commits since mid August, I've observed the following:

1. `Moon.component` is now `Moon.extend`
2. instead of `el`, it is `root` when creating a moon app
3. instead of the `m-model` attribute, it is now named `m-bind`
4. Any plugins or other things that access the internal attributes like `$children` now use a version without `$`
5. SVG stuff now just is called `SVG` instead of `isSVG`
6. To say that an element is text type, it was `TEXT_TYPE`, but now the shorter literal `"#text"` is used

There's a couple things that I don't quite know about, like `dynamic` and `functional` when it comes to component things.

What other bumps are there in transitioning?

**Comment by kbrsh** [comment]
Well, you're right about most of them, except number 6. I simply removed a constant called `TEXT_TYPE`. It was just an alias for `"#text"`.

Some other things are:

1. You can mount a component with the constructor:
    ```js
    new MyComponent({
      root: "#target",
      props: {}
    });
    ```
2. The `m-static` directive
3. No support for custom key codes

**Comment by LeviSchuck** [comment]
It looks like `callMethod` was removed too.
I have to do `instance.get('methodhere')(params);`

**Comment by kbrsh** [comment]
Ahh yes. This is still being worked on and I'm thinking of more ways to make it efficient. I'm thinking maybe:

```
// Can be called anything
instance.methods.foo();

// Cannot conflict with internal methods/properties
instance.foo();
```

**Comment by Bampye** [comment]
And `m-html` has gone too.

**Comment by kbrsh** [comment]
Yup @Bampye. It was just syntactic sugar for setting a DOM property. You can now manually set it with:

```html
<div m-literal:innerHTML="rawHTML"></div>
```

**Comment by kbrsh** [comment]
Another thing: `.set` can now accept an object.

```js
// Shallow merge with data and notifies observer
instance.set({
  foo: true,
  bar: [1, 2, 3]
});
```

**Comment by rizkyramadhan** [comment]
It seems component slot is breaking. 
When I put `<slot></slot>` in component template, it just render `<slot></slot>`.

**Comment by kbrsh** [comment]
@rizkyramadhan They were replaced by an insert system (to not collide with web components). So slots are now replaced with `<m-insert>` elements. Right now there is no named inserts (so no `<m-insert name="foo">`), but I might add it later.

**Comment by russelg** [comment]
In regards to that `.set` change, it should probably be mentioned the old keypath method (i.e `.set('obj.prop.nested')`) does not work anymore.

**Comment by kbrsh** [comment]
DOM properties are now set with a new directive: `m-dom`. It's a replacement for `m-literal:prop.dom`. Also, static elements are marked with a flag along with being hoisted.

These both were added in e6208979ff29f45584c92238871b8123c1bf26f8.
